docs/package-management.md

```diff
@@ -267,3 +267,127 @@ The advantage of this approach —as opposed to vendoring the JavaScript package
+]
+```
+
+To use a library from the installed package, add the library name to the `dune` file under the `libraries` field. For example, if our project structure looks like:
+
+<pre class="text-ocaml"><code class="language-text hljs plaintext">project_name/
+├── _opam
+├── src
+│   ├── dune
+│   ├── ReactComponent1.ml
+│   ├── ReactComponent2.ml
+│   └── lib
+│        ├── dune
+│        └── data.ml
+├── dune-project
+├── dune
+├── package.json
+└── ...</code></pre>
+<pre class="text-reasonml"><code class="language-text hljs plaintext">project_name/
+├── _opam
+├── src
+│   ├── dune
+│   ├── ReactComponent1.re
+│   ├── ReactComponent2.re
+│   └── lib
+│        ├── dune
+│        └── data.re
+├── dune-project
+├── dune
+├── package.json
+└── ...</code></pre>
+
+then `react-reason` should be added to the `dune` file under the `src` folder:
+
+```text
+(melange.emit
+ (target output)
+ (alias react)
+ (libraries lib reason-react)
+ (preprocess
+  (pps reactjs-jsx-ppx))
+ (module_systems es6))
+```
+
+(Some packages, like `reason-react`, will need to be preprocessed using a ppx, which may also need to be installed via opam.)
+
+### Unpublished opam packages
+
+Some packages may have an `.opam` file, but they are not yet published in the official opam repository. These packages should be installed using `opam pin`, e.g., `opam pin add melange-fetch.dev https://github.com/melange-community/melange-fetch`. The project `.opam` file should then be updated in two places:
+
+```text
+...
+depends: [
+  ...
+  "melange-fetch" {dev}
+]
+pin-depends: [
+  [ "melange-fetch.dev" "git+https://github.com/melange-community/melange-fetch" ]
+]
+```
+
+Once installed, the libraries included in the package can be added to the `dune` file:
+
+```text
+(melange.emit
+ (target output)
+ (alias react)
+ (libraries lib reason-react melange-fetch)
+ (preprocess
+  (pps reactjs-jsx-ppx))
+ (module_systems es6))
+```
+
+### NPM packages
+
+A number of Melange compatible packages can be found on npm. Many older, but still useful, compatible BuckleScript libraries can be found on npm, e.g., `bs-json`. An npm package is installed like any normal npm package is: `npm install @glennsl/bs-json`, which will add the package to the `package.json` file at the root of our project.
+
+In order to get Dune to pick up and process the npm package so that it can be consumed in our application, the `dune` file at the root of our project must be updated with a `subdir` stanza:
+
+```text
+(subdir
+ node_modules
+ (vendored_dirs @glennsl)
+ (dirs @glennsl)
+ (subdir
+  @glennsl
+  (subdir
+   bs-json
+   (subdir
+    src
+    (library
+     (name bs_json)
+     (wrapped false)
+     (modes melange))))))
+```
+
+If the `dune` file contains the line `(dirs :standard \ node_modules)`, it should be removed.
+Note the `(vendored_dirs @glennsl)` field in the stanza. This is used to silence warnings from vendored libraries, keeping the terminal output relevant to our application code.
```

Comment:
this is wrong in the example we give because we're not defining a public library in the vendored directory.